### PR TITLE
Fix #306750, Fix#306751 - Chord alignment

### PR DESCRIPTION
**Pull request by njvdberg**
Fix #306750 - Chord symbol alignment incorrectly forces alignment between above & below chords and across staves
Fix #306751 - Chord symbol alignment changes default position without reason

Resolves https://musescore.org/en/node/306750
Resolves https://musescore.org/en/node/306751

Extended the algorithm so it aligns chord symbols, fret diagrams or RNA symbols per staff by collecting the symbols per staff and run the alignment for every staff. When looking for the reference for the alignment, the algorithm now handles placement above and below the staff as two different cases. This solves #306750.

Removed an extra offset which was not required anymore but was a left-over of an earlier
version of the algorithm. This solves #306751.

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/Documentation/blob/master/WorkflowAndGuidelines/CodeGuidelines.md)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by MarcSabatella** [comment]
Works advertised, code looks good to me!  vtest failure looks like slight diff not sure why, but perhaps a result of the change to the default?

**Comment by njvdberg** [comment]
The vtest failure is expected. It shows a difference in <code>harmony-15</code> which is a test which I include in PR #6060 to test the chord alignment. This difference is the expected result of this PR in order to solve [#306751](https://musescore.org/en/node/306751).
A new reference is added.

**Comment by anatoly-os** [comment]
@njvdberg rebase, please.

File: libmscore/layout.cpp [hunk 1]
```diff
@@ -3309,74 +3309,75 @@ void alignHarmonies(const System* system, const std::vector<Segment*>& sl, bool
...
-                  system->staff(e->staffIdx())->skyline().add(h->shape().translated(h->pos() + fd->pos() + s->pos() + m->pos()));
+            // Add all aligned objects to the sky line.
+            for (Element* e : modified) {
+                  const Segment* s = toSegment(e->parent());
+                  const MeasureBase* m = toMeasureBase(s->parent());
+                  system->staff(e->staffIdx())->skyline().add(e->shape().translated(e->pos() + s->pos() + m->pos()));
+                  if (e->isFretDiagram()) {
+                        FretDiagram* fd = toFretDiagram(e);
+                        Harmony* h = fd->harmony();
+                        system->staff(e->staffIdx())->skyline().add(h->shape().translated(h->pos() + fd->pos() + s->pos() + m->pos()));
```
Comment (Jojo-Schmitz) [inline libmscore/layout.cpp:3379]: this here is where the merge conflict should be due to my fix in  #6236

**Comment by njvdberg** [comment]
@anatoly-os Rebase done.

**Comment by MarcSabatella** [comment]
Is there any progress on the issue discussed on Telgram, where disabling autoplace does not exclude chord symbols from the alignment?  This still feels like a necessary change.

**Pull request merged**